content/blogposts/2023/guest-post-arafat-khan-php-python.md

```diff
@@ -0,0 +1,256 @@
+---
+title: "Climbing Mount Everest in Flip-flops: My PHP journey as a Python dev"
+authors:
+  - name: Arafat Khan
+    url: https://twitter.com/arafatkatze
+  
+publishDate: 2023-11-21T10:00-01:00
+description: "When tasked with creating a printing API in PHP despite having zero experience, Arafat Khan leveraged an AI coding assistant named Cody to transform what felt like an Everest climb into a swift 3 day project that resulted in a fully functioning API."
+tags: [blog]
+slug: "guest-post-arafat-khan-php-python"
+published: true
+heroImage: https://storage.googleapis.com/sourcegraph-assets/blog/guest-post-arafat-khan-php-python-og.png
+socialImage: https://storage.googleapis.com/sourcegraph-assets/blog/guest-post-arafat-khan-php-python-og.png
+---
+
+<Alert type="secondary">This is a guest blog post from our community. <a href="https://discord.com/servers/sourcegraph-969688426372825169" target="_blank">Join our Discord</a> to meet the author and connect with other developers passionate about Code Search and AI!</Alert>
+
+_Ever thought about climbing Mount Everest in flip-flops?_ That’s how I felt diving into _PHP_ for the first time.
+
+A few weeks ago, a professional endeavor landed on my desk creating a printing API. The goal was to build a library in PHP that lets developers easily play with commands to scan and print documents programmatically. With a functional Python API version already in my arsenal, I was initially optimistic about the task. Translating the code to another language, PHP in this case, seemed like a straight path. However, the catch was my complete lack of familiarity with PHP, making the experience seem like a herculean effort at first glance. To add to the challenge, I imposed a time constraint on myself so now I had to learn a new language and build a full-blown project that I could share with the world.
+
+This task entailed a couple of different things
+
+* Building and publishing an entire PHP package with easy install instructions and simple commands to print/scan documents while cleanly hiding all the API complexity behind some classes.
+* Build comprehensive documentation for those keen on exploring edge cases.
+* Add decent testing/error handling to maintain code and take future developments into account.
+
+Ultimately, my goal was to offer this functionality to PHP users
+
+<Figure
+  src="https://storage.googleapis.com/sourcegraph-assets/blog/guest-post-arafat-khan-php-python/image_1.png"
+  caption="Rough Outline of the Print API"
+/>
+
+While I knew that I knew nothing about PHP I also knew that I had help that I could rely on. I started with the only two things that I had at the time:
```

Comment:
While I knew that I knew nothing about PHP I also knew that I had help that I could rely on. I started with the only two things that I had at the time: => While I knew that I knew nothing about PHP, I also knew that I had help that I could rely on. I started with the only two things that I had at the time: